### PR TITLE
[AGENT-4706] Implement Airflow Operators to create Custom Model

### PR DESCRIPTION
**Pull request by oleksandr-saienko**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Added CreateCustomInferenceModelOperator and CreateCustomModelVersionOperator implementaion
Added example DAG

## Rationale
users should be able to created Custom Model using Airflow Operators

### CODE REVIEW

**Comment by oleksandr-saienko** [comment]
Jenkins run

**Comment by oleksandr-saienko** [comment]
Jenkins run

**Comment by oleksandr-saienko** [comment]
Jenkins run

File: datarobot_provider/operators/custom_models.py [hunk 1]
```diff
@@ -189,3 +189,171 @@ def execute(self, context: Dict[str, Any]) -> str:
...
+        )
+
+        custom_model_description = (
+            context["params"].get("custom_model_description", None)
+            if self.description is None
+            else self.description
+        )
+
+        custom_model = dr.CustomInferenceModel.create(
+            name=custom_model_name,
```
Comment (alexromanyuk) [inline datarobot_provider/operators/custom_models.py:246]: will it fail with NPE if name is None?

File: tests/unit/operators/test_custom_models.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
+# Copyright 2023 DataRobot, Inc. and its affiliates.
```
Comment (alexromanyuk) [inline tests/unit/operators/test_custom_models.py:1]: would be nice to add negative test cases to check behavior when user missing some required input parameters

**Review by alexromanyuk — APPROVED** [review]
:+1: looks good, just minor comments

File: datarobot_provider/operators/custom_models.py (hunk 1 above)
Comment (oleksandr-saienko) [reply datarobot_provider/operators/custom_models.py:246]: Added check for required params

File: datarobot_provider/operators/custom_models.py [hunk 3]
```diff
@@ -189,3 +189,171 @@ def execute(self, context: Dict[str, Any]) -> str:
...
+            if self.custom_model_folder is None
+            else self.custom_model_folder
+        )
+
+        custom_model_version = dr.CustomModelVersion.create_clean(
+            custom_model_id=self.custom_model_id,
+            training_dataset_id=self.training_dataset_id,
+            base_environment_id=self.base_environment_id,
+            holdout_dataset_id=self.holdout_dataset_id,
+            folder_path=folder_path,
```
[the remark this answers is not in the document]
Comment (oleksandr-saienko) [reply datarobot_provider/operators/custom_models.py:343]: Added check for required params

File: tests/unit/operators/test_custom_models.py (hunk 2 above)
Comment (oleksandr-saienko) [reply tests/unit/operators/test_custom_models.py:1]: Added negative test cases

**Pull request merged**